### DISCUSSION TITLE
[HarmonyOS]Using CustomFontManagerImpl class causes Chinese characters not to display

### DISCUSSION THREAD

**Discussion by CeSun** [AvaloniaUI/Avalonia#17781] [Q&A]
I am adapting Avalonia to a new platform and I am currently facing a problem where the default font manager cannot find the default font and causes a crash, so I used the CustomFontManagerImpl class and now I can successfully display text on HarmonyOS, but only English characters can be displayed.

![QQ_1734336174192](https://github.com/user-attachments/assets/03593564-6ca7-4b18-9748-39f9a9261cfb)
![QQ_1734336197018](https://github.com/user-attachments/assets/f7a80df9-8af8-45fa-b9ff-ab4a7c0da46d)
https://github.com/CeSun/OpenHarmony.Avalonia/blob/main/Src/Avalonia.OpenHarmony/CustomFontManagerImpl.cs
https://github.com/CeSun/OpenHarmony.Avalonia/blob/main/Src/Avalonia.OpenHarmony/OpenHarmonyPlatform.cs#L22

**Comment by kyuranger** [comment]
Avalonia has not officially supported HarmonyOS yet. The existing support for HarmonyOS from Avalonia is from third parties, so I think it is likely that the problem cannot be solved (or can only be solved by modifying the Avalonia source code).

**Reply by rabbitism** [reply]
He is the third party.

**Reply by thevortexcloud** [reply]
Indeed, they are trying to add support for HarmonyOS to Avalonia.

https://github.com/CeSun/OpenHarmony.Avalonia

**Comment by Gillibald** [comment]
You should no longer replace the font manager, but register a custom system font collection. Skia's font manager will not work on HarmonyOS so your font manager can't rely on the system.
